### PR TITLE
ORC-844: Improve hashCode Methods

### PR DESCRIPTION
**Pull request by belugabehr**
### What changes were proposed in this pull request?
Auto-generate some hashCode methods.

### Why are the changes needed?
Better hash code values for better distributions; less code.

### How was this patch tested?
No changes to functionality. Use existing unit tests.

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
Looks reasonable.

cc @pgaref , @wgtmac , @williamhyun

**Review by wgtmac — APPROVED** [review]
Looks good to me as well.

**Comment by dongjoon-hyun** [comment]
Thank you, @wgtmac . Merged to main for Apache ORC 1.8.0.

File: java/core/src/java/org/apache/orc/TypeDescription.java [hunk 1]
```diff
@@ -380,13 +380,14 @@ public TypeDescription clone() {
 
   @Override
   public int hashCode() {
-    long result = category.ordinal() * 4241 + maxLength + precision * 13 + scale;
-    if (children != null) {
```
Comment (dongjoon-hyun) [inline java/core/src/java/org/apache/orc/TypeDescription.java:384]: ORC-1291 is reported due to the removal of this.

Comment (dongjoon-hyun) [reply java/core/src/java/org/apache/orc/TypeDescription.java:384]: I made a fix for this.
- https://github.com/apache/orc/pull/1283

**Pull request merged**